File: TechBlazers-Part1/SavingsAccount.java
public class SavingsAccount {

    public static double calculateInitialDeposit(double finalAccountValue, double monthlyInterestRate, int years) {
        double annualInterestRate = (monthlyInterestRate / 100) / 12;
        int noOfMonths = years * 12;
        double initialDepositAmount = finalAccountValue / Math.pow(1 + annualInterestRate, noOfMonths);
        return initialDepositAmount;
    }

    public static void main(String[] args) {
        double finalAccountValue = 10000;
        double monthlyInterestRate = 5;
        int years = 10;

        double initialDepositAmount = calculateInitialDeposit(finalAccountValue, monthlyInterestRate, years);
        System.out.printf("The initial deposit amount is: %.2f%n", initialDepositAmount);
    }
}

File: TechBlazers-Part1/savings_account.py
def savings_account(final_account_value, monthly_interest_rate, years):
	annual_interest_rate = (monthly_interest_rate / 100) / 12
	no_of_months = years * 12
	initial_deposit_amount = final_account_value / (1 + annual_interest_rate) ** no_of_months
	return initial_deposit_amount

print(savings_account(10000, 5, 10))